### DISCUSSION TITLE
[Feature] Public Link improvements

### DISCUSSION THREAD

**Discussion by chriexpe** [immich-app/immich#3170] [Feature Request]
### The feature

The current implementation does work and has some good options, but it lacks some polishing and other features such as:

- [ ] Share button when the photo/video is opened (currently you can only do it by selecting it in the "panel").

- [ ] Add sharing option on mobile.

- [ ] When only a single item is shared, it would be better if it would open in full screen only (currently the link loads to a page that list the shared items).

- [ ] URL Shortener, this can either be from e.g. bitly or even the selfhosted options such as Yourls (also allowing the user to default "always use URL shortener").

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Most of these are now implemented
